src/WebInterface.js

```diff
@@ -14,92 +15,113 @@ const Config = require('./Config');
 class WebInterface {
   
   constructor(state) {
-    this.state = state;
-    this.port  = Config.get('webPort') || 9000;
+    this.state  = state;
+    this.port   = Config.get('webPort') || 9000;
+    this.secretFile = __dirname + '/../.ranvierSecret';
   }
 
   init() {
+    util.log('[WEB] Initializing...');
+    this.generateSecret();
+    this.setUpGetRoutes();
 
-    const message = { message: "RanvierMUD API Endpoints: /npcs /players /items /rooms /help" };
-    // Routes for the API's GET response.
-    router.get('/',        (req, res) => res.json(message));
-    router.get('/npcs',    this.getResponseData(this.state.MobFactory, 'npcs'));
-    router.get('/players', this.getResponseData(this.state.PlayerManager, 'players'));     
-    router.get('/items',   this.getResponseData(this.state.ItemManager, 'items'));
-    router.get('/rooms',   this.getResponseData(this.state.RoomManager, 'rooms'));
-    router.get('/help',    this.getResponseData(this.state.HelpManager, 'helps'));
-
-    router.post('/authenticate', (req, res) => {
-      const accountName = req.body.name;
-      const account = this.state.AccountManager.findByName(accountName);
-      
-      if (!account) {
-        util.log(`[WEB] ${accountName} Auth Failed: Account not found.`);
-        return res.json({ success: false, message: `${accountName} was not found.`});
-      }
+    // Authentication route.
+    router.post('/authenticate', (req, res) => this.authenticate(req, res));
```

Comment:
Right. So here's where I was looking for feedback -- I am wondering if only POST routes (to make changes) should be authenticated, or if all of them should, or what. Maybe it could be configurable in `ranvier.json`. 

Though, it probably makes sense to just protect all of the routes.

Also, I realized we are currently doing nothing with the "Secret" besides using it to make a token.